src/map/utils/battleutils.cpp

```diff
@@ -6984,73 +6969,16 @@ namespace battleutils
 
             if (PCoverAbilityUser != nullptr)
             {
-                float coverAbilityTargetX = PCoverAbilityTarget->loc.p.x;
-                float coverAbilityTargetZ = PCoverAbilityTarget->loc.p.z;
-                float mobX                = PMob->loc.p.x;
-                float mobZ                = PMob->loc.p.z;
-
-                float xdif       = coverAbilityTargetX - mobX;
-                float zdif       = coverAbilityTargetZ - mobZ;
-                float slope      = 0;
-                float maxSlope   = 0;
-                float minSlope   = 0;
-                bool  zDependent = true; // using a slope where z is dependent var
-
-                if (abs(xdif) <= abs(zdif))
-                {
-                    slope = xdif / zdif;
-
-                    float angle = (float)atan((double)1) * 2 - atan(slope);
-
-                    float zoffset   = cos(angle) / 2;
-                    float xoffset   = sin(angle) / 2;
-                    float maxXpoint = mobX + xoffset;
-                    float maxZpoint = mobZ - zoffset;
-                    float minXpoint = mobX - xoffset;
-                    float minZpoint = mobZ + zoffset;
-
-                    maxSlope = ((maxXpoint - coverAbilityTargetX) / (maxZpoint - coverAbilityTargetZ));
-                    minSlope = ((minXpoint - coverAbilityTargetX) / (minZpoint - coverAbilityTargetZ));
-
-                    zDependent = false;
-                }
-                else
-                {
-                    slope = zdif / xdif;
-
-                    float angle = (float)atan((double)1) * 2 - atan(slope);
+                // using same variable names as trick attack function, for consistent understanding
+                uint8 angleTAmob = worldAngle(PMob->loc.p, PCoverAbilityUser->loc.p);
+                float distTAmob  = distanceSquared(PCoverAbilityUser->loc.p, PMob->loc.p);
```

Comment:
I tend to use `distanceSquared` if i just need to compare two distances. I only used sqrt once to match the style that cover originally had (but i've since changed that to `distTAmob <= static_cast<float>(PMob->GetMeleeRange() * PMob->GetMeleeRange())`)

Reply:
Putting performance considerations to the side for a moment (`distance` is just `sqrt(distanceSquared(...)`, and `sqrt` hasn't been an expensive operation since the 90's):

The issue I have with `distanceSquared` is that it abstracts measurements in yalms into yalms^2, and it introduces the need to continually multiply out the number you're comparing to. We have a `constexpr square(...)` function, but if the numbers being brought in aren't known at compiletime you lose the benefit of the `constexpr`.

At the cost of a couple more `sqrt` calls you can use `distance` and you wouldn't need to multiply `PMob->GetMeleeRange()` with itself and the code would be more intuitive. Since this is a complex and mathsy area of code, I'd lean towards readability over pure performance (especially when there's a sea of stupid things we do elsewhere that make optimisation of float comparisons irrelevant).

Reply:
Understood, I'll update to use distance everywhere